merge.go

```diff
@@ -282,12 +282,23 @@ func (m *Merger) mergeArray(orig []interface{}, n []interface{}, node string) []
 // the array
 func (m *Merger) mergeArrayDefault(orig []interface{}, n []interface{}, node string) []interface{} {
 	DEBUG("%s: performing index-based array merge", node)
-	if err := canKeyMergeArray("original", orig, node, "name"); err == nil {
-		if err := canKeyMergeArray("new", n, node, "name"); err == nil {
+	var err error
+	if err = canKeyMergeArray("original", orig, node, "name"); err == nil {
+		if err = canKeyMergeArray("new", n, node, "name"); err == nil {
 			return m.mergeArrayByKey(orig, n, node, "name")
 		}
 	}
 
+	//Warn the user about any unintuitive behavior that may have gotten us here.
+	if warning, isWarning := err.(WarningError); isWarning && warning.HasContext(eContextDefaultMerge) {
+		mergeStratStr := "inline"
+		if m.AppendByDefault {
+			mergeStratStr = "append"
+		}
+		warning.SetWarning(ansi.Sprintf("%s\n  @Y{Falling back to %s merge strategy}", warning.Error(), mergeStratStr))
+		warning.Warn()
```

Comment:
Seems weird to receive an error that you then have to set the message value on, and then log. Why not just call something like 
```
if err != nil {
    mergeStratStr := "inline"
    if m.AppendByDefault {
        mergeStratStr = "append"
    }
    log.Warn("Could not merge by key: %s\n@y{Falling back to %s merge strategy}", err, mergeStratStr)
}
```